lib/mayaUsd/fileio/shading/shadingModeDisplayColor.cpp

```diff
@@ -304,40 +325,86 @@ DEFINE_SHADING_MODE_IMPORTER(displayColor, context)
     const GfVec3f transparencyColor =
         UsdMayaColorSpace::ConvertLinearToMaya(linearTransparency);
 
-    std::string shaderName(_tokens->MayaShaderName.GetText());
+    // We default to lambert if no conversion was requested:
+    const TfToken& shadingConversion
+        = jobArguments.shadingConversion != UsdMayaShadingConversionTokens->none
+        ? jobArguments.shadingConversion
+        : UsdMayaShadingConversionTokens->lambert;
+    std::string shaderName(shadingConversion.GetText());
     SdfPath shaderParentPath = SdfPath::AbsoluteRootPath();
     if (shadeMaterial) {
         const UsdPrim& shadeMaterialPrim = shadeMaterial.GetPrim();
         shaderName =
             TfStringPrintf("%s_%s",
                 shadeMaterialPrim.GetName().GetText(),
-                _tokens->MayaShaderName.GetText());
+                shadingConversion.GetText());
         shaderParentPath = shadeMaterialPrim.GetPath();
     }
 
-    // Construct the lambert shader.
-    MFnLambertShader lambertFn;
-    MObject shadingObj = lambertFn.create();
-    lambertFn.setName(shaderName.c_str());
-    lambertFn.setColor(
-        MColor(displayColor[0], displayColor[1], displayColor[2]));
-    lambertFn.setTransparency(
-        MColor(transparencyColor[0], transparencyColor[1], transparencyColor[2]));
-    
-    // We explicitly set diffuse coefficient to 1.0 here since new lamberts
-    // default to 0.8. This is to make sure the color value matches visually
-    // when roundtripping since we bake the diffuseCoeff into the diffuse color
-    // at export.
-    lambertFn.setDiffuseCoeff(1.0);
-
-    const SdfPath lambertPath =
-        shaderParentPath.AppendChild(TfToken(lambertFn.name().asChar()));
-    context->AddCreatedObject(lambertPath, shadingObj);
-
-    // Find the outColor plug so we can connect it as the surface shader of the
-    // shading engine.
-    MPlug outputPlug = lambertFn.findPlug("outColor", &status);
-    CHECK_MSTATUS_AND_RETURN(status, MObject());
+    // Construct the selected shader.
+    MObject           shadingObj;
+    UsdMayaTranslatorUtil::CreateShaderNode(
+              MString(shaderName.c_str()),
+              shadingConversion.GetText(),
+              UsdMayaShadingNodeType::Shader,
+              &status,
+              &shadingObj);
+    if (status != MS::kSuccess) {
+        // we need to make sure assumes those types are loaded..
+        TF_RUNTIME_ERROR(
+            "Could not create node of type '%s' for prim '%s'.\n",
+            shadingConversion.GetText(),
+            primSchema.GetPath().GetText());
+        return MObject();
+    }
+
+    MPlug outputPlug;
+#if MAYA_API_VERSION >= 20200000
+    if (shadingConversion != UsdMayaShadingConversionTokens->standardSurface) {
+#endif
+        MFnLambertShader lambertFn;
+        lambertFn.setObject(shadingObj);
+        lambertFn.setName(shaderName.c_str());
```

Comment:
Is the `setName()` here redundant now because we've already created the node with this name above?